src/mlpack/methods/ann/activation_functions/hardshrink_function.hpp

```diff
@@ -0,0 +1,97 @@
+/**
+ * @file hardshrink_function.hpp
+ * @author Lakshya Ojha
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_HARD_SHRINK_FUNCTION_HPP
+#define MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_HARD_SHRINK_FUNCTION_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <algorithm>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * HardShrinkage operator is defined as,
+ *         | x , if x >  lambda
+ *  f(x) = | x , if x < -lambda
+ *         | 0, otherwise
+ *
+ * lambda is set to 0.5 by default.
+ *
+ *          | 1 , if x >  lambda
+ *  f'(x) = | 1 , if x < -lambda
+ *          | 0 , otherwise
+ */
+class HardShrinkFunction
+{
+ public:
+  /**
+   * Computes the Hard Shrinkage function.
+   *
+   * @param x Input data.
+   * @return f(x).
+   */
+  static double Fn(const double x, const double lambda = 0.5)
```

Comment:
We should implement this as a layer, similar to: https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/ann/layer/elu.hpp#L127 because there is no way, a user can use that function with a base layer and adjust the lambda parameter.

Reply:
Hi @zoq as I can see it can be done. But it would have been better if we were aware of this before when the pr had started. None the less I will analyse and learn here how to implement it as you have asked.
Thanks.